README.md

```diff
@@ -1,3 +1,210 @@
 # Sample Metadata
 
 An API to manage sample + other related metadata.
+
+## Environment variables
+
+Sample-metadata API uses environment variables to manage it's configuration.
+
+### Running the web server
+
+The default behaviour for the web server is to:
+
+- Use a GCP secret for the DB credentials
+
+You can configure this with:
+
+```shell
+# use credentials defined in db/python/connect.py:dev_config
+export SM_ENVIRONMENT=dev
+# use specific mysql settings
+export SM_DEV_DB_PROJECT=sm_dev
+export SM_DEV_DB_USER=root
+export SM_DEV_DB_PORT=3307
+export SM_DEV_DB_HOST=127.0.0.1
+export SM_DEV_DB_PASSWORD=root
+export SM_ENVIRONMENT=dev
+```
+
+### Using the Python API
+
+The default behaviour for the installable Python API is to:
+
+- Use the production server
+- With a service account
+
+You can configure these separately with:
+
+```shell
+# use localhost:8000
+export SM_ENVIRONMENT=dev
+ # use a local account to generate identity-token for auth
+export SM_USE_SERVICE_ACCOUNT=false
+```
+
+
+## Database
+
+Sample-metadata DB uses MariaDB (for the system-versioned-tables) to store metadata.
+
+There are two dbs you'll need to initialise:
+
+- `sm_admin`: Stores project-independent metadata (like the internal_sample_id to project map)
+- `sm_<project>`: Stores project specicic information.
+
+In development, we recommend `sm_dev`, as [there is a default dev_config](https://github.com/populationgenomics/sample-metadata/blob/8b122453d1cd26c09966b8e54909bf712da5263e/db/python/connect.py#L78-L85).
+
+Schema is managed with _liquibase_ with the `project.xml` and `global.xml` changelogs. To use liquibase, you'll need to download the `mariadb` driver:
+
+```shell
+pushd db/
+wget https://repo1.maven.org/maven2/org/mariadb/jdbc/mariadb-java-client/2.7.2/mariadb-java-client-2.7.2.jar
+popd
+```
+
+### Using Maria DB docker image
+
+Pull mariadb image
+
+```bash
+docker pull mariadb
+```
+
+Run a mariadb container that will server your database. `-p 3307:3306` remaps the port to 3307 in case if you local MySQL is already using 3306
+
+```bash
+docker stop mysql-p3307  # stop and remove if the container already exists
+docker rm mysql-p3307
+docker run -p 3307:3306 --name mysql-p3307 -e MYSQL_ROOT_PASSWORD=root -d mariadb
+```
+
+Initialize databases (you may need to enter the password on each command).
+
+```bash
+mysql --host=127.0.0.1 --port=3307 -u root -p -e 'CREATE DATABASE sm_dev;'
+mysql --host=127.0.0.1 --port=3307 -u root -p -e 'CREATE DATABASE sm_admin;'
+mysql --host=127.0.0.1 --port=3307 -u root -p -e 'show databases;'
+```
+
+Download [`mariadb-java-client-2.7.2.jar`](https://repo1.maven.org/maven2/org/mariadb/jdbc/mariadb-java-client/2.7.2/mariadb-java-client-2.7.2.jar) into the `db/` folder.
+
+Then create the database schemas.
+
+```bash
+pushd db/
+liquibase update --url jdbc:mariadb://127.0.0.1:3307/sm_admin --username=root --password=root --classpath mariadb-java-client-2.7.3.jar --changelog-file=global.xml
```

Comment:
I think this is supposed to be 
`liquibase --url jdbc:mariadb://127.0.0.1:3307/sm_admin --username=root --password=root --classpath mariadb-java-client-2.7.3.jar --changeLogFile global.xml update`

Reply:
Thanks, I've fixed this in the latest commit.